src/main/java/io/netty/incubator/codec/http3/Http3FrameToHttpObjectCodec.java

```diff
@@ -187,18 +187,20 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
                 Http3Headers headers = HttpConversionUtil.toHttp3Headers(last.trailingHeaders(), validateHeaders);
                 promise = writeWithOptionalCombiner(ctx,
                         new DefaultHttp3HeadersFrame(headers), promise, combiner, true);
-            }
-            if (!readable) {
+            } else if (!readable) {
+                // Release the data and just use EMPTY_BUFFER. This might allow us to give back memory to the allocator
+                // faster.
                 last.release();
```

Comment:
`last.release()` is now not called when `hasTrailers && !readable`. i think this could leak the buffer.

Reply:
Good catch @yawkat ... PTAL again